src/components/playback/playbackmanager.js

```diff
@@ -1933,58 +1896,109 @@ class PlaybackManager {
                 } else if (firstItem.Type !== 'BoxSet') {
                     sortBy = 'SortName';
                 }
-                promise = getItemsForPlayback(serverId, mergePlaybackQueries({
+
+                return getItemsForPlayback(serverId, mergePlaybackQueries({
                     ParentId: firstItem.Id,
                     Filters: 'IsNotFolder',
                     Recursive: true,
                     // These are pre-sorted
                     SortBy: sortBy,
                     MediaTypes: 'Audio,Video'
                 }, queryOptions));
-            } else if (firstItem.Type === 'Episode' && items.length === 1 && getPlayer(firstItem, options).supportsProgress !== false) {
-                promise = new Promise(function (resolve, reject) {
-                    const apiClient = ServerConnections.getApiClient(firstItem.ServerId);
+            }
+        }
 
-                    apiClient.getCurrentUser().then(function (user) {
-                        if (!user.Configuration.EnableNextEpisodeAutoPlay || !firstItem.SeriesId) {
-                            resolve(null);
-                            return;
+        async function getSeriesOrSeasonPlaybackPromise(firstItem) {
+            const apiClient = ServerConnections.getApiClient(firstItem.ServerId);
+
+            const episodesResult = await apiClient.getEpisodes(firstItem.SeriesId || firstItem.Id, {
+                IsVirtualUnaired: false,
+                IsMissing: false,
+                UserId: apiClient.getCurrentUserId(),
+                Fields: ['Chapters', 'Trickplay']
+            });
+            const originalResults = episodesResult.Items;
+            const isSeries = firstItem.Type === 'Series';
+            let foundItem = false;
+            episodesResult.Items = episodesResult.Items.filter(function (e) {
+                if (foundItem) {
+                    return true;
+                }
+
+                if (!e.UserData.Played && (isSeries || e.SeasonId === firstItem.Id)) {
+                    foundItem = true;
+                    return true;
+                }
+
+                return false;
+            });
+            if (episodesResult.Items.length === 0) {
+                if (isSeries) {
+                    episodesResult.Items = originalResults;
+                } else {
+                    episodesResult.Items = originalResults.filter(function (e_1) {
```

Comment:
Is it necessary to rename `e`?

Reply:
Not sure how this got changed... Reverted it back to just `e`